staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -245,6 +221,18 @@ func (r *withRetry) Before(ctx context.Context, request *Request) error {
 		return nil
 	}
 
+	// At this point we've made atleast one attempt, post which the response
+	// body should have been fully read and closed in order for it to be safe
+	// to reset the request body before we reconnect, in order for us to reuse
+	// the same TCP connection.
+	if seeker, ok := request.body.(io.Seeker); ok && request.body != nil {
+		if _, err := seeker.Seek(0, io.SeekStart); err != nil {
+			err = fmt.Errorf("failed to reset request body for %T - %w", request, err)
+			r.trackPreviousError(err)
```

Comment:
Calling trackPreviousError seems suspect here - the error is "we can't read the body" but that's not the previous error anymore?

Reply:
the current error has `EOF` wrapped, so the higher level function should still be able to determine that the error returned (current error) is an EOF error.
We should already have a test for that, if not we should add one.

Reply:
As a client, I'm not 100% sure that a failure to seek from a programmer error (passing request a non-startable stream) is an error that should be returned.  However, implementing Seekable is probably as close as we can get to inferring that intent, in which case the "current" error does become "we failed to seek due to a filesystem or other remote error", so I guess this is ok.